bgpd/bgp_vty.c

```diff
@@ -3167,9 +3168,12 @@ DEFUN (no_neighbor,
 			}
 
 			other = peer->doppelganger;
+			peer_notify_unconfig(peer);
 			peer_delete(peer);
-			if (other && other->status != Deleted)
+			if (other && other->status != Deleted) {
+				peer_notify_unconfig(peer);
```

Comment:
We are sending notification for "doppelganger" peer. So need to change it to peer_notify_unconfig(other);